### ISSUE TITLE
Modular Losses

### ISSUE THREAD

**Issue by nmichlo** [nmichlo/disent#11]
Losses currently have to be defined at a framework level through the use of overrides.

It would be nice if losses can be built directly, for example:

```python3
beta_vae_loss =  4 * KlRegLoss() + MseRecLoss()
# or even
beta_vae_loss =  Param(cfg, 'beta') * KlRegLoss() + MseRecLoss()

# computing the loss for a training step as
loss, logs = beta_vae_loss.compute_loss(ds_posterior, ds_prior, zs_sampled, xs_partial_recon, xs_targ)
```

```python3
dfc_vae_loss =  Param(cfg, 'beta') * KlRegLoss() + 0.5 * MseRecLoss() + 0.5 * DfcRecLoss()

# computing the loss for a training step as
loss, logs = dfc_vae_loss.compute_loss(ds_posterior, ds_prior, zs_sampled, xs_partial_recon, xs_targ)
```

Alternative data structures should be discussed.

This would require a substantial rewrite, however existing frameworks
could define these loss components in the constructor.

**Comment by nmichlo** [comment]
Too much work to implement at the moment.
